### ISSUE TITLE
Add security_group_id attribute to aws_directory_service_directory resource when using 'type = "ADConnector"'

### ISSUE THREAD

**Issue by wendtek** [terraform-providers/terraform-provider-aws#7466]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating an AD Connector, Microsoft Managed AD, or Simple AD in AWS Directory Service, it will create an associated security group. When using the `aws_directory_service_directory` resource you must specify a `type`. When specifying either `SimpleAD` or `MicrosoftAD`, it will provide an attribute called `security_group_id` containing the security group. I am requesting that the same behavior is implemented for `type = "ADConnector"`

### New or Affected Resource(s)

*aws_directory_service_directory

### Potential Terraform Configuration

```hcl
resource "aws_directory_service_directory" "ad-connector" {
  name     = "mydomain.local"
  password = "AdministratorForADPassword"
  size     = "Small"
  type     = "ADConnector"

  connect_settings {
    customer_username = "AdministratorForAD"
    vpc_id            = "${data.terraform_remote_state.vpc.vpc_id}"

    customer_dns_ips = [
      "172.0.0.1",
      "172.0.0.2",
    ]

    subnet_ids = [
      "${element(data.terraform_remote_state.vpc.vpc_subnet_private_ids,0)}",
      "${element(data.terraform_remote_state.vpc.vpc_subnet_private_ids,1)}",
    ]
  }
}

resource "aws_security_group_rule" "egress-to-radius-server" {
  type                     = "egress"
  from_port                = 1812
  to_port                  = 1812
  protocol                 = "udp"
  security_group_id        = "${aws_directory_service_directory.ad-connector.security_group_id}"
  source_security_group_id = "${aws_security_group.radius-server.id}"
}

```

**Comment by bflad** [comment]
Support for setting the `security_group_id` attribute when `type = "ADConnector"` has been merged and will release with version 1.59.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
